CommunityToolShedMvc/Controllers/ToolController.cs

```diff
@@ -0,0 +1,192 @@
+                    new SqlParameter("@ID", communityid)
+                );
+
+            var viewModel = new ToolPersonCommunityConditions(conditionTypes, person, community);
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        public ActionResult Add(int communityid, ToolPersonCommunityConditions viewModel)
+        {
+            if (ModelState.IsValid)
+            {
+                var id = DatabaseHelper.ExecuteScalar<int>(@"
+                        BEGIN TRAN;
+
+                        INSERT INTO Tool (OwnerID, Name, ConditionID, Warnings)
+                        VALUES (@OwnerID, @Name, @ConditionID, @Warnings);
+
+                        DECLARE @ToolID int;
+                        SET @ToolId = cast(scope_identity() as int);
+
+                        INSERT INTO Shed (ToolID, CommunityID)
+                        VALUES (@ToolID, @CommunityID);
+
+                        SELECT @CommunityID;
+
+                        COMMIT TRAN;
+                    ",
+                        new SqlParameter("@OwnerID", ((CustomPrincipal)User).Person.Id),
+                        new SqlParameter("@Name", viewModel.Tool.Name),
+                        new SqlParameter("@ConditionID", viewModel.Tool.ConditionID),
+                        new SqlParameter("@Warnings", viewModel.Tool.Warnings),
+                        new SqlParameter("@CommunityID", communityid)
+                    );
+
+                return RedirectToRoute("Default", new { controller = "Community", action = "Overview", id = communityid });
+            }
+
+            List<ConditionType> conditionTypes = DatabaseHelper.Retrieve<ConditionType>(@"
+                    SELECT ID, Name
+                    FROM Condition
+                    ORDER BY ID
+                ");
+
+            viewModel.SetConditionTypes(conditionTypes);
+            return View(viewModel);
+        }
+
+
+        public ActionResult Edit(int communityid, int id)
+        {
+            List<ConditionType> conditionTypes = DatabaseHelper.Retrieve<ConditionType>(@"
+                    SELECT ID, Name
+                    FROM Condition
+                    ORDER BY ID
+                ");
+
+            Person person = DatabaseHelper.RetrieveSingle<Person>(@"
+                    SELECT CONCAT(FirstName,' ',LastName) AS FullName
+                    FROM Person
+                    WHERE ID = @ID
+                ",
+                    new SqlParameter("@ID", ((CustomPrincipal)User).Person.Id)
+                );
+
+            Community community = DatabaseHelper.RetrieveSingle<Community>(@"
+                    SELECT Name, ID
+                    FROM Community
+                    WHERE ID = @ID
+                ",
+                    new SqlParameter("@ID", communityid)
+                );
+
+            var viewModel = new ToolPersonCommunityConditions(conditionTypes, person, community);
+
+            viewModel.Tool = DatabaseHelper.RetrieveSingle<Tool>(@"
```

Comment:
Maybe update the `ToolPersonCommunityConditions` view model's constructor to accept a `Tool` model instance?